lib/signon/client.rb

```diff
@@ -76,5 +104,9 @@ def do_request(request, uri)
         http.request(request)
       end
     end
+
+    def already_exists(res)
+      res.code == "400" && JSON.parse(res.body).dig("error") == "Record already exists"
```

Comment:
This is provided by Signon here: https://github.com/alphagov/signon/blob/7bab43419a1e9340273e8fb2e69d2de8b72da2f4/app/helpers/admin_api_helper.rb#L27. It's only used by this client so we can easily change it.

Would a response like `{ "error": "ApplicationAlreadyExistsError"}` feel more robust rather than `{"error": "Record already exists"}`?

If not through parsing the error message, perhaps we can use a `409 Conflict` status code instead. `400` is super broad, which is the reason one must parse the error message. What do you think?

Reply:
Ah, that's good news then. I reckon either 409 or a more stable-looking message would be an improvement.

Reply:
Cool, I'll modify the API and add that in a separate commit.